combit.ListLabel.SchemaAwareJsonDataProvider/SchemaAwareJsonDataProvider.cs

```diff
@@ -35,6 +38,12 @@ public string Schema
                 _schemaLocation = value;
                 var fileProvider = SchemaFileProvider ?? Options?.FileProvider ?? new NetworkFileProvider();
                 _schema = JsonSchema.FromJsonAsync(fileProvider.ReadAsString(_schemaLocation)).Result;
+
+                RootTableName = _schema.Title ?? RootTableName;
```

Comment:
This may break existing projects that rely on the "old" logic of retrieving the table name. However, as this provider is mostly used in source code form and the addition of a compatibility switch would be trivial, I support this change although it might be breaking.

Reply:
I added this, because one cannot set the RootTableName from JsonSchema info without loading the JsonSchema from outside. So maybe we can make the JsonSchema property public and then remove the new RootTableName and ArrayValueName setters.

Reply:
I think your implementation has advantages in keeping the data schema information where it belongs. And if someone should have a problem with this change, they may easily add a compat switch to keep the "old" behavior.